Gems/Atom/RPI/Code/Include/Atom/RPI.Public/RPIUtils.h

```diff
@@ -75,17 +75,20 @@ namespace AZ
 
-        //! Retrieve a region of image pixel values (float) for specified mip and slice
+        //! Process a region of image pixel values (float) for specified mip and slice
+        //! The visitor function `callback` is invoked on each pixel specified by the region
         //! NOTE: The topLeft coordinate is inclusive, whereas the bottomRight is exclusive
-        bool GetSubImagePixelValues(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<float> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
+        bool GetSubImagePixelValues(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::function<void(const AZ::u32& x, const AZ::u32&y, const float& value)> callback, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
 
-        //! Retrieve a region of image pixel values (uint) for specified mip and slice
+        //! Process a region of image pixel values (uint) for specified mip and slice
+        //! The visitor function `callback` is invoked on each pixel specified by the region
         //! NOTE: The topLeft coordinate is inclusive, whereas the bottomRight is exclusive
-        bool GetSubImagePixelValues(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<AZ::u32> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
+        bool GetSubImagePixelValuesUint(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::function<void(const AZ::u32& x, const AZ::u32& y, const AZ::u32& value)> callback, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
 
-        //! Retrieve a region of image pixel values (int) for specified mip and slice
+        //! Process a region of image pixel values (int) for specified mip and slice
+        //! The visitor function `callback` is invoked on each pixel specified by the region
         //! NOTE: The topLeft coordinate is inclusive, whereas the bottomRight is exclusive
-        bool GetSubImagePixelValues(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::span<AZ::s32> outValues, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
+        bool GetSubImagePixelValuesInt(const AZ::Data::Asset<AZ::RPI::StreamingImageAsset>& imageAsset, AZStd::pair<uint32_t, uint32_t> topLeft, AZStd::pair<uint32_t, uint32_t> bottomRight, AZStd::function<void(const AZ::u32& x, const AZ::u32& y, const AZ::s32& value)> callback, uint32_t componentIndex = 0, uint32_t mip = 0, uint32_t slice = 0);
```

Comment:
nit: I think these functions are the point where they should be split to several lines.

Reply:
Thanks you are right. I'll clean these up and the others that have gotten a little long too (and in the cpp as well)